src/domain/dataset-update-flow/src/aggregates/update.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{DateTime, Utc};
+use event_sourcing::*;
+use kamu_task_system::TaskID;
+use opendatafabric::DatasetID;
+
+use crate::*;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Aggregate, Debug)]
+pub struct Update(Aggregate<UpdateState, (dyn UpdateEventStore + 'static)>);
+
+impl Update {
+    /// Creates a dataset update process
+    pub fn new(update_id: UpdateID, dataset_id: DatasetID, trigger: UpdateTrigger) -> Self {
+        Self(
+            Aggregate::new(
+                dataset_id.clone(),
+                UpdateInitiated {
+                    event_time: Utc::now(),
```

Comment:
Would be nice to externalize time sources from the start

Reply:
Externalized here and in Tasks aggregate, but in the form of an argument.
The dependency on the time source service should be established in services creating domain objects.